### PR TITLE
Add tests related to loans and return.

### LINKED ISSUE #201: [PE-D][Tester D] Incorrect error message when input for Loan command is wrong.
As shown here the error message displayed is "Unable to parse as date" which is not related to the command.

-------------
Labels: `type.FeatureFlaw` `severity.Low`
original: sherylkong18/ped#6

### PR DESCRIPTION
**Pull request by sprintaway**
Add tests for loancommandparsertest.

Add tests for loancommandtest.

Add complete range of keywords that work for loancommand in UG.

Fix issues assigned to me.

Add ReturnCommandTest

Add ReturnCommandParserTest

Resolves #201, #195, #191

### CODE REVIEW

**Comment by sprintaway** [comment]
There is a weird issue that occurs which causes some tests (specifically in deletecommandparser) to fail at random.

File: docs/UserGuide.md [hunk 1]
```diff
@@ -155,18 +155,42 @@ Examples:
...
 
-Format: `loan USER_INDEX BOOK_INDEX [DUE DATE]`
+Format: `loan USER_INDEX BOOK_INDEX [DUE_DATE]`
 
 * Loans the book to some user at their respective specified `INDEXES`.
 * The indexes refer to the index number shown in the displayed user and book list respectively.
 * The indexes **must be a positive integer** 1, 2, 3, …​
 * The respective specified `INDEXES` **must be present in their lists**.
-* The books that are loaned out will appear at the top of the book list.
+* The books that are loaned out will appear at the top of the book list after closing and reopening of app.
```
Comment (parth-io) [inline docs/UserGuide.md:164]: 'after closing and reopening of app' --> 'after relaunching the app' may be better

File: docs/UserGuide.md [hunk 2]
```diff
@@ -155,18 +155,42 @@ Examples:
...
+Date formats that will work:
+dd/MM/yyyy, yyyy-MM-dd, next {DAY}, following {DAY}, last {DAY}, past {DAY}, yesterday, today, tomorrow, (any positive integer) {DAY} later,
+(any positive integer) {DAY} ago.
+</div>
+
+<div markdown="block" class="alert alert-info">
+
+**:information_source: Notes about the command format:**<br>
+
+* Next strictly refers to next week while following refers to the upcoming weekday/weekend.
```
Comment (parth-io) [inline docs/UserGuide.md:183]: Do we wanna add a comparison between `this {DAY}` and `next {DAY}`, given that users expect `this {DAY}` to return today's date if today is the `{DAY}`

File: src/test/java/bookface/logic/commands/LoanCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+    @Test
+    public void execute_invalidIndexUnfilteredPersonList_throwsCommandException() {
+        Index outOfBoundIndex = Index.fromOneBased(model.getFilteredPersonList().size() + 1);
+        LoanCommand loanCommand = new LoanCommand(outOfBoundIndex, TypicalIndexes.INDEX_FIRST_BOOK);
+        CommandTestUtil.assertCommandFailure(loanCommand, model, Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+    }
+
+
+    // For this, it shows "Book is already loaned out." instead of MESSAGE_LOAN_SUCCESS, but I'm not sure why this
```
Comment (parth-io) [inline src/test/java/bookface/logic/commands/LoanCommandTest.java:43]: Is this fixed?

File: src/test/java/bookface/logic/commands/ReturnCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+import bookface.commons.core.index.Index;
+import bookface.model.Model;
+import bookface.model.ModelManager;
+import bookface.model.UserPrefs;
+import bookface.testutil.TypicalIndexes;
+
+public class ReturnCommandTest {
+    private final Model model = new ModelManager(getAllTypicalBookFaceData(), new UserPrefs());
+
+    //This test is very weird, if i test it on its own, it works, but if i test all the tests at once, it fails.
```
Comment (parth-io) [inline src/test/java/bookface/logic/commands/ReturnCommandTest.java:21]: Same

File: src/test/java/bookface/logic/commands/delete/DeleteUserCommandTest.java [hunk 5]
```diff
@@ -46,6 +47,8 @@ public void execute_invalidIndexUnfilteredList_throwsCommandException() {
         CommandTestUtil.assertCommandFailure(deleteUserCommand, model, Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
     }
 
+    //Displays error "Person cannot be deleted; there are loans that are not settled!" but why?
```
Comment (parth-io) [inline src/test/java/bookface/logic/commands/delete/DeleteUserCommandTest.java:50]: Same

File: src/test/java/bookface/logic/commands/LoanCommandTest.java (hunk 3 above)
Comment (sprintaway) [reply src/test/java/bookface/logic/commands/LoanCommandTest.java:43]: For the tests, sorry I'm still unable to fix them. But can pin the blame on me if asked who's in charge of loan and return tests so I dont think team will be penalized for this.

File: src/test/java/bookface/logic/commands/ReturnCommandTest.java (hunk 4 above)
Comment (sprintaway) [reply src/test/java/bookface/logic/commands/ReturnCommandTest.java:21]: As above

File: src/test/java/bookface/logic/commands/delete/DeleteUserCommandTest.java (hunk 5 above)
Comment (sprintaway) [reply src/test/java/bookface/logic/commands/delete/DeleteUserCommandTest.java:50]: As above

File: docs/UserGuide.md (hunk 1 above)
Comment (sprintaway) [reply docs/UserGuide.md:164]: Fixed

File: docs/UserGuide.md (hunk 2 above)
Comment (sprintaway) [reply docs/UserGuide.md:183]: `this {DAY}` is not stated in the command format which means it isnt supported, so if users enter `this {DAY}` then we can claim that its coming in the future and is out of scope. 

In addition there's no real reason to use `this {DAY}` if today is the `{DAY}`, users will simply use `today`

Comment (parth-io) [reply docs/UserGuide.md:183]: Ah yea, so it falls under compound/complex then

File: src/test/java/bookface/logic/commands/LoanCommandTest.java (hunk 3 above)
Comment (parth-io) [reply src/test/java/bookface/logic/commands/LoanCommandTest.java:43]: Oh I guess we can merge first then - we can figure this out later tonight or tomorrow during the meeting

**Review by parth-io — APPROVED** [review]

**Review by parth-io — APPROVED** [review]

**Pull request merged**